spec/unit/puppet/provider/powershell_input_spec.rb

```diff
@@ -0,0 +1,47 @@
+#! /usr/bin/env ruby
+require 'spec_helper'
+require 'puppet/type'
+require 'puppet/type/base_dsc'
+
+example_dsc_resource =Puppet::Type.type(:dsc_file)
+
+describe 'powershell_provider' do
+
+  let :dsc_file do
+    Puppet::Type.type(:dsc_file).new(
+      :name => 'fio',
+      :dsc_ensure          => 'Present',
+      :dsc_destinationpath => 'C:/test.file',
+    )
+  end
+
+  describe "powershell provider" do
+
+    before(:each) do
+      @provider = example_dsc_resource.provider(:powershell).new(dsc_file)
+    end
+
+    describe "when quotes are present" do
```

Comment:
Will need an option where quotes are not present. 

Otherwise mostly good. I'm curious if there is a better way to get to the provider. :)